### DISCUSSION TITLE
Dynamic parallel queries with SWR (equivalent of useQueries)

### DISCUSSION THREAD

**Discussion by gfox1984** [vercel/swr#1038] [General]
Hey, 

I have an array of user IDs and I need to make a request for each of them using SWR.

I'm looking for an equivalent to the [useQueries hook](https://react-query.tanstack.com/guides/parallel-queries#dynamic-parallel-queries-with-usequeries) in React Query.

How do you achieve the same thing with SWR (each user should be cached individually)?

**Comment by ErnestBrandi** [comment]
Hi, you can just fetch a list of users _cached as a whole_ (based on that ids array).

**Reply by gfox1984** [reply]
Yeah but a few issues with that:

- I have a basic jsonapi that cannot load multiple users at once (yes, I could pass a fetcher that makes multiple requests and returns Promise.All)
- Elsewhere in the code, if I request data about a single user that's in that list, I want it to be returned from cache

**Reply by ErnestBrandi** [reply]
Ok, so you can just do more _composition_ 
```
function User({ id }) {
  const { data } = useSWR(`api/users/${id}`, fetcher);
  // ...
}

export default function IndexPage() {
  const ids = [1, 3];

  return (
    <>
      {ids.map((id) => (
        <User key={id} id={id} />
      ))}
    </>
  );
}
```
Full example here : https://codesandbox.io/s/musing-chatterjee-c2upx?file=/pages/index.js

**Reply by gfox1984** [reply]
That's a good suggestion. Won't help when you need to read data about all the fetched users at the composition root though.

**Reply by ErnestBrandi** [reply]
Then use another hook for all users as I suggested in the first place OR add an array in the root component that you update in every `User` when data is fetched.

**Reply by kafkahw** [reply]
@gfox1984 there's a workaround discussed here that might work for you: https://github.com/vercel/swr/discussions/786#discussioncomment-1436230

**Reply by kafkahw** [reply]
@ErnestBrandi I think this is a valid feature request because sometimes we are limited by imperfect APIs (where are not controlled by us). In this case, if API provides a fetch all endpoint that accepts an array of user id, the problem can be easily solved with your suggested approach (i.e. `use another hook for all users`). unfortunately, it's not always that case.
Therefore, if useSWR can provide an equivalent of `userQueries`, it'll make devs life a lot easier when dealing with parallel/multiple requests.

**Reply by gfox1984** [reply]
I have to say that I already logged this feature request a while ago: https://github.com/vercel/swr/issues/1041 . Since then, I've made the switch to [react-query](https://react-query.tanstack.com/) as it was more tailored to my use case.

**Reply by rodbs** [reply]
Looking forwards to this feature too!!! I'd like to fetch several items at the same time (send an array), but later if a particular item needs to be fetched  not fetch it if it's already in cache.
